_posts/2024-06-10-infinispan-15-queries.adoc

```diff
@@ -0,0 +1,317 @@
+---
+layout: blog
+title: Infinispan 15 indexing & query news
+permalink: /blog/:year/:month/:day/infinispan-15-queries
+date: '2024-06-10T00:00:00.000-00:00'
+author: fax4ever
+tags: [ "search", "vector", "score", "knn", "indexing", "embeddings" ]
+---
+
+= Infinispan 15 indexing & query news
+
+The Infinispan team is happy to announce some improvements on query APIs that are already present on Infinispan 15:
+
+1. <<rest_projections,Rest queries with projections>>: more projection types are supported using the REST API.
+2. <<continuous-queries,Query cache API>>: regular and continuous queries can be defined directly from the cache API.
+3. <<filter-elements,Filter elements for kNN queries>>: it is now possible to filter out the set of entities on which to apply a kNN-vector search.
+4. <<index-query-by-keys, Index by keys - query by keys>>: indexes can be defined also on the keys, not only on the values, so that we can query also the keys values.
+
+[[rest_projections]]
+== Rest queries with projections
+
+Recently we introduced score and versions projections, that along with entity and field projections are the all
+kinds of projections we support with the Infinispan queries.
+
+The same projections are now available using the REST query API.
+
+For instance if we have an entity like the following:
```

Comment:
I would group entity, query and result... something like `Following an example of entity, query and result:`